flows/actions/base.go

```diff
@@ -42,12 +42,12 @@ func resolveGroups(run flows.FlowRun, step flows.Step, action flows.Action, refe
 			// group is an expression that evaluates to an existing group's name
 			evaluatedGroupName, err := excellent.EvaluateTemplateAsString(run.Environment(), run.Context(), ref.NameMatch)
 			if err != nil {
-				log = append(log, events.NewErrorEvent(err))
+				*log = append(*log, events.NewErrorEvent(err))
 			} else {
 				// look up the set of all groups to see if such a group exists
 				group = groupSet.FindByName(evaluatedGroupName)
 				if group == nil {
-					log = append(log, events.NewErrorEvent(fmt.Errorf("no such group with name '%s'", evaluatedGroupName)))
+					*log = append(*log, events.NewErrorEvent(fmt.Errorf("no such group with name '%s'", evaluatedGroupName)))
```

Comment:
These always feel a bit dirty to me, I'd rather see us return the log than passing in pointers to lists. I think that's a bit more typical in go anyways.